frontend/src/composables/use-floating.ts

```diff
@@ -7,10 +7,10 @@ import { useEventListener } from "@vueuse/core";
 /** use floating-ui to position dropdown */
 
 export const useFloating = (
-  anchor: Ref<HTMLElement>,
-  dropdown: Ref<HTMLElement>,
+  anchor: Ref<HTMLElement | undefined>,
+  dropdown: Ref<HTMLElement | undefined>,
   fit = false,
-) => {
+): { calculate: () => Promise<void>; style: Ref<CSSProperties> } => {
```

Comment:
The explicit return type shouldn't be necessary here, ts should be able to infer it. If it is necessary, something is wrong.